keps/sig-storage/2848-transfer-volumesnapshots-namespaces/README.md

```diff
@@ -0,0 +1,801 @@
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+This KEP proposes a method for transferring VolumeSnapshot between namespaces. 
+
+Currently, VolumeSnapshot is bound to specific namespace, and VolumeSnapshotContent is Cluster scoped. The binding relationship between these two storage resources is one-to-one.
+This proposal targets allowing users to conveniently move these resources without manual creation
+of each independent object (VolumeSnapshot, VolumeSnapshotContent).
+
+Based on community design sessions, and inspired by [API for PVC Namespace Transfer](https://github.com/kubernetes/enhancements/pull/2110/files), the following is being proposed :-
+
+- `StorageTransferRequest` resources are created in the source namespace.
+- `StorageTransferApproval` resources are created in the target namespace.
+-  When a controller detects matching request/approval resources, a transfer in initiated.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+Allow Kubernetes users to quickly and conveniently transfer VolumeSnapshots across namespaces.
+### Goals
+- Use CRDs for VolumeSnapshot Namespace Transfer request/approval.
+- Edit external SnapshotController to accept/reject/init VolumeSnapshot transfer.
+- The target VolumeSnapshot will point to a newly created VolumeSnapshotContent.
+- Use Validation Webhook to reject any transfer contains secret.
+- Options to be introduced allowing transfer of VolumeSnapshot with secret for specific CSI Drivers (e.g. Ceph).
```

Comment:
@xing-yang @jsafrane @bswartz I added this as we discussed in the meeting.

Reply:
@Elbehery FYI, here are a few approaches for sharing secrets across namespaces in other projects:
https://github.com/vmware-tanzu/carvel-secretgen-controller/blob/develop/docs/secret-export.md
https://kyverno.io/policies/other/sync_secrets/

Reply:
@xing-yang thanks you so much for this, I will dig after it and roll back my findings and how we can use it here 👍🏽 🙏🏽